languages/rust/oso/tests/test_polar.rs

```diff
@@ -357,3 +370,47 @@ fn test_macros() {
 
     test.qvar_one(r#"new Baz().world() = x"#, "x", "goodbye world".to_string());
 }
+
+// TODO: dhatch see if there is a relevant test to port.
+#[test]
+fn test_unify_externals() {
+    let mut test = OsoTest::new();
+
+    #[derive(PartialEq, Clone, Debug)]
```

Comment:
I think we should use the derive macro in most cases for tests, since that's likely the main way people will use the library.

Reply:
I'm not sure.  I like the explicitness (also in this case I haven't implemented anything to derive the equality function using the proc macro.

Reply:
Also feel the test surface area is smaller if it doesn't use the proc macro and uses this interface directly.